lib/config.js

```diff
@@ -0,0 +1,101 @@
+const cds = require('@sap/cds')
+const { camelToSnake } = require('./util')
+
+/**
+ * Makes properties of an object accessible in both camelCase and snake_case.
+ * Snake_case gets precedence over camelCase.
+ * @template T
+ * @param {T} target - The object to proxy.
+ * @returns {T} - The proxied object.
+ */
+const camelSnakeHybrid = target => {
+    // @ts-expect-error - expecting target to be of type {}, which is not T (same for following)
+    const proxy = new Proxy(target, {
+        get(target, prop) {
+            // @ts-expect-error
+            return target[camelToSnake(prop)] ?? target[prop]
+        },
+        set(target, p, v) {
+            // @ts-expect-error
+            target[camelToSnake(p)] = v
+            return true
+        }
+    })
+    // need to make sure all properties are initially available in snake_case
+    // @ts-expect-error
+    for (const [k,v] of Object.entries(target)) {
+        // @ts-expect-error
+        proxy[k] = v
+    }
+    // @ts-expect-error
+    return proxy
+}
+class Config {
+    static #defaults = {
+        propertiesOptional: true,
+        useEntitiesProxy: false,
+        inlineDeclarations: 'flat'
+    }
+
+    constructor() {
+        this.values = {...Config.#defaults, ...(cds?.env?.typer ?? {})}
```

Comment:
`cds.env` should never be null/undefined.

```suggestion
        this.values = {...Config.#defaults, ...(cds.env.typer ?? {})}
```

Also, `cds.env` uses `process.cwd()`, so are we sure we run in the project root?
Otherwise we'd need to use `cds.env.for('cds', <projectRoot>)`

Reply:
> there is no guarantee cds-typer is started from the project root. How would we find `<projectRoot>`?

That's a classic question for which we don't have a good answer.  Often this is done by finding a package.json w/ sap/cds as a dependency, or a .cdsrc.json.
But before we go there, I wonder why this hasn't been an issue so far.  How is the location determined for typer in VS Code?  Otherwise `cds-typer '*'` would not work.

In CIs I guess the project location is correctly set.

Reply:
Yes, we should settle on a standard way to find a project root given a file.
But let's do that in a separate PR.

> we apply a naive heuristic going up the a file structure until we find a directory that looks like a root directory.

I thought there is already an implementation in cds-typer to do that which we could use.  But I guess this is vscode-cds then?

Reply:
> I thought there is already an implementation in cds-typer to do that which we could use. But I guess this is vscode-cds then?

Correct, this is currently part of cds-lsp.